File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major = nums[0];
        int count = 0;

        for(auto&num:nums){
          
          if(count ==0){
            major = num;
          }

          count += (num == major) ? 1 : -1;



        }


        return major;
        
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        ans = nums[0]

        for i in range(1,len(nums)):
            if count == 0:
                ans = nums[i]
                count =1 
            else:
                if ans == nums[i]:
                    count +=1
                else:
                    count -=1
        return ans
        